kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterContext.java

```diff
@@ -92,10 +92,48 @@ public interface FilterContext {
      * @param <T> The type of the response
      * @return CompletionStage that will yield the response.
      * @see io.kroxylicious.proxy.filter Thread Safety
+     * @deprecated use {@link FilterContext#sendRequest(RequestHeaderData, ApiMessage)} instead.
      */
     @NonNull
+    @Deprecated(forRemoval = true, since = "0.3.0")
     <T extends ApiMessage> CompletionStage<T> sendRequest(short apiVersion, @NonNull ApiMessage request);
 
+    /**
+     * Send a request from a filter towards the broker.   The response to the request will be
+     * made available to the filter asynchronously, by way of the {@link CompletionStage}.  The
+     * CompletionStage will contain the response header and response body, or null of the
+     * request does not have a response.
+     * <h4>Header</h4>
+     * <p>The caller is required to provide a {@link RequestHeaderData}. It is recommended that the
+     * caller specify the {@link RequestHeaderData#requestApiVersion()}. This can be done conveniently
+     * with forms such as:</p>
+     * <pre>{@code new RequestHeaderData().setRequestApiVersion(4)}</pre>
+     * <p>The caller may also provide a {@link RequestHeaderData#clientId()} an
+     * {@link RequestHeaderData#unknownTaggedFields()}. Kroxylicious will automatically set the
+     * {@link RequestHeaderData#requestApiKey()} to be consistent with the {@code request}.
+     * {@link RequestHeaderData#correlationId()} is ignored.</p>
+     * <h4>Filtering</h4>
+     * <p>The request will pass through all filters upstream of the filter that invoked the operation,
+     * invoking them, but not itself.  Similarly, the response will pass through all filters upstream
```

Comment:
```suggestion
     * invoking them.  Similarly, the response will pass through all filters upstream
```

Not invoking itself is inherent in the definition of upstream, isn't it?